### DISCUSSION TITLE
Conditional nested write

### DISCUSSION THREAD

**Discussion by abdel-ships-it** [prisma/prisma#11369] [Q&A] [answered]
Im trying to perform a conditional nested write, where I'm performing a `connect` if certain data is present. But the code looks real ugly and I was wondering if there is another way to do this. I have thought about using the `$transaction` API but I can't really do that because the connected records need to know the `id` of the created user.

I think that the [`interactiveTransactions`](https://www.prisma.io/docs/concepts/components/prisma-client/transactions#interactive-transactions-in-preview) could solve my problem here but I saw people advice against it because of potential deadlocks and it being a last resort measure. So my question is, is there a cleaner way to do this? 

```typescript
this.$prisma.user.create({
  data: {
    name: createUserDto.name,
    email: createUserDto.email,
    license: {
      ...(createUserDto.licenseId
        ? {
            connect: {
              id: createUserDto.licenseId,
            },
          }
        : {}),
    }
});

```

**Comment by austincrim** [comment, marked as answer]
Hey @realappie 👋 ,

The code you posted above is the simplest way to do a conditional `connect`. If the code doesn't feel "clean", you could extract the ternary to its own variable.

```ts
const license = createUserDto.licenseId ? { connect: { id: createUserDto.licenseId }} : {}
this.$prisma.user.create({
  data: {
    name: createUserDto.name,
    email: createUserDto.email,
    license
});
```

If you have ideas on a cleaner API for this scenario, could you open a feature request with the details?

Thanks so much!

**Reply by abdel-ships-it** [reply]
Ah that's a bummer, extracting it out is what I would have usually done but then I miss out on type completion 😅. I will give this some thought and perhaps open a feature request in the future, thanks for your help!

**Comment by ridwansameer** [comment]
I wonder if there's a way to make this cleaner.

We currently have a table that could EITHER have a quiz_id `connect`, a lesson_id `connect` OR a autogenerated_quiz `CREATE`. So in this scenario we have to do something like this


```js
const { topic_id, module_name, description, photo_url, approved_status, module_contents } = req.body;
    const module = await prisma.module.create({
      data: {
        module_name,
        description,
        photo_url,
        topic_id,
        approved_status: "incomplete",
        module_content: {
          create: module_contents.map((content: any): Prisma.module_contentCreateInput => ({
            lesson: {
              ...(content.lesson_id
                ? {
                  connect: {
                    id: content.lesson_id,
                  },
                }
                : {}),
            },
            quiz: {
              ...(content.quiz_id
                ? {
                  connect: {
                    id: content.quiz_id,
                  },
                }
                : {}),
            },
            autogenerated_quiz: {
              ...(content.autogenerated_quiz_name ? {
                create: {
                  autogenerated_quiz_name: content.autogenerated_quiz_name,
                  autogenerated_quiz_type: content.autogenerated_quiz_type,
                  num_easy: 1,
                  num_med: 1,
                  num_hard: 1,
                  autogenerated_quiz_subtopic: {
                    create: content.subtopic_ids.map((subtopic_id: number): Prisma.autogenerated_quiz_subtopicUpdateWithoutAutogenerated_quizInput => ({
                      subtopic: {
                        connect: {
                          id: subtopic_id
                        }
                      }
                    }))
                  }
                }
              } : {})
            },
          }))
        }
      },
      include: {
        module_content: {
          include: {
            autogenerated_quiz: true,
          }
        }
      }
    })
```

Attention to the `lesson:`, `quiz:` and `autogenerated_quiz:` part. Have to do conditionals for all three.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
